Codecademy/codepedia
# Codepedia: Restructure how entries are loaded

# Overview
This PR updates the file structure of Codepedia to be language-based instead of concept based.

## Specific Changes
- Entries are now expected to be placed in folders as `codepedia/entries/[language name]/[concept].md`
- Entries now have explicit meta-details defined for them in `codepedia/entries/[language name]/_meta/[concept].json`
- A set of helpers were added to `node-helpers` to automate some metadata related tasks
- Documentation was updated to reflect the new file structure
- Codepedia loads appropriately from the new file structure
- Searching / filtering on Codepedia now shows a more relevant view based on the filter
- When filtering by text, keywords and language are displayed as tags on the cards.
- Commit hooks were added to compile the meta details and compiled GH page on commit

## Notes
For code review, I'd suggest filtering out Codepedia entries (`.md` and `.json` files; content didn't change, just where they are located). Make sure to take a look at the README and the contribution guide before filtering out .md files though.

## Testing Instructions
1. Run the application locally (instructions in the README)
2. Try adding a new entry via `yarn new-entry`
3. Make a change in a `_meta` file and run `yarn build:map` to see the change
4. Try filtering based on keyword, title, or description
